frontend/main.go

```diff
@@ -125,36 +129,63 @@ func httpFileServerWith404(fs http.FileSystem) http.Handler {
 	})
 }
 
-func main() {
-	flags := parseFlags()
-
-	if flags.Version {
-		fmt.Printf("version.Info{Version:'%s', GitCommit:'%s', BuildDate:'%s'}\n", version.OdigosVersion, version.OdigosCommit, version.OdigosDate)
-		return
-	}
-
-	// Load destinations data
-	err := destinations.Load()
-	if err != nil {
-		log.Fatalf("Error loading destinations data: %s", err)
-	}
-
-	// Connect to Kubernetes
-	err = initKubernetesClient(&flags)
-	if err != nil {
-		log.Fatalf("Error creating Kubernetes client: %s", err)
-	}
-
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
 
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
+func main() {
+    flags := parseFlags()
+
+    if flags.Version {
+        fmt.Printf("version.Info{Version:'%s', GitCommit:'%s', BuildDate:'%s'}\n", version.OdigosVersion, version.OdigosCommit, version.OdigosDate)
+        return
+    }
+
+    // Load destinations data
+    err := destinations.Load()
+    if err != nil {
+        log.Fatalf("Error loading destinations data: %s", err)
+    }
+
+    // Create a channel to capture Ctrl+C
+    interruptChan := make(chan os.Signal, 1)
+    signal.Notify(interruptChan, os.Interrupt, syscall.SIGINT)
+
+    go func() {
+        for range interruptChan {
+            log.Println("Received Ctrl+C. Shutting down gracefully...")
+            // Implement any cleanup or shutdown logic here
+
+            os.Exit(0)
+        }
+    }()
+
+    // Attempt to start the server on the specified or default port
+    maxPortAttempts := 10
+    for {
+        r, err := startHTTPServer(&flags)
+        if err != nil {
+            log.Printf("Error starting server on port %d: %s", flags.Port, err)
+
+            if flags.Port < defaultPort+maxPortAttempts {
```

Comment:
This code assumes `flags.Port` is initialized to `defaultPort`. 
But the user can run `odigos ui --port 4000` or `odigos ui --port 2000` and then you would run the loop once or many many times.